synapse/lib/urlhelp.py

```diff
@@ -85,4 +85,28 @@ def sanitizeUrl(url):
     Note: this depends on this being a reasonably-well formatted URI that starts with a scheme (e.g. http) and '//:'
     Failure of this condition yields the original string.
     '''
-    return _url_re.sub(r'\g<front>****', url)
+    try:
+        info = chopurl(url)
+    except Exception as e:
+        return url
+    else:
+        if info.get('passwd'):
+            # Rebuild the URL from info
+            valu = f"{info.get('scheme')}://{info.get('user')}:****@"
+            host = info.get('host')
+            port = info.get('port')
+            if ':' in host and port:
```

Comment:
This just gets used for logging purposes right? I think checking whether the original URL used brackets around an IPv6 would be most accurate since they could have used brackets even without specifying a port, but if it is just for display clarity here it probably doesn't matter.

Reply:
Yup - its mainly for logging / display purposes; typically in error handling. That is why its memoized here, so we aren't re-parsing urls over and over again. We should make the memoize change regardless of how we change this function.

The re-bracketing is from how `chopurl` handles IPv6 hosts with port numbers.

We could also move to a regex that is more explicit about identifying a prefix scheme / optional user group with an optional password group; and then just replace the password group. I was trying to move away from an regex though.

Reply:
Yeah I think it's fine as is